draft-ietf-cdni-ci-triggers-rfc8007bis.xml

```diff
@@ -1218,568 +1044,686 @@ uCDN-A                         dCDN-B                         dCDN-C
+   </t>
 
-   <t><list style="empty" hangIndent="3">
-   <t><list style="hanging" hangIndent="3">
-   
-      <t hangText="Name: action">
-   <vspace blankLines="1"/>
-   Description: Defines the type of the CI/T Trigger Action.
-   <vspace blankLines="1"/>
-   Value: Trigger Action Type, as defined in <xref target="trigger-action"/>.
-   <vspace blankLines="1"/>
-   Mandatory: Yes.
+   <t>
+	This document also defines an initial set of trigger subject values and registers them in 
+	the IANA "CDNI CI/T Trigger Subjects" registry:
    </t>
+      
+   <texttable style="full"><ttcol align="left"> JSON String</ttcol>
+   <ttcol align="left"> Description</ttcol>
+   <c>metadata</c>
+   <c>Indicating the trigger target specification refers to Metadata object(s), as defined at <xref target="RFC8006"/>.</c>
+   <c>content</c>
+   <c>Indicating the trigger target specification refers to client-facing content objects.</c>
+   </texttable>
+   </section>
 
-   <t hangText="Name: specs">
+   <section anchor="urls-spec" title="URLs Spec">
+   <t>
+      The "urls" spec type allows uCDN to manage uCDN content or metadata objects
+      held by dCDN based on the objects' URLs.
+        </t>
+
+   <t>The URLs spec is encoded as a JSON object containing the following name/value pairs:</t>
+
+        <t><list style="empty" hangIndent="3">
+        <t><list style="hanging" hangIndent="3"><t hangText="Name: urls">
+        <vspace blankLines="1"/>
+   Description: An array of URLs over which the trigger MUST be executed.
+        <vspace blankLines="1"/>
+   Value: A JSON array of URLs represented as JSON strings.
+        <vspace blankLines="1"/>
+        Mandatory: Yes.
+        </t>
+
+        <t hangText="Name: url-type">
+        <vspace blankLines="1"/>
+   Description: Type of URL used.
+        <vspace blankLines="1"/>
+   Value: URL Type as defined in <xref target="url-types"/>.
+        <vspace blankLines="1"/>
+        Mandatory: No. When omitted or empty, "published" URL type is assumed.
+        </t>
+
+        </list></t>
+        </list></t>
+
+
+        <t>Below is an example of a JSON-serialized URLs spec object, matching the metadata at 
+        metadata.example.com/a/b/c.
+        </t>
+<sourcecode type="json" name="urls-spec-json"><![CDATA[
+{
+  "trigger-subject": "metadata",
+  "generic-trigger-spec-type": "urls",
+  "generic-trigger-spec-value": {
+    "urls": [ "https://metadata.example.com/a/b/c" ],
+    "url-type": "published"
+  }
+}
+       ]]></sourcecode>
+      </section>
+
+
+
+      <section anchor="ccids-spec" title="CCIDs Spec">
+        <t> The "ccids" spec type allows uCDN to specify the
+        Content Collection IDentifier (CCID) of content to which the trigger applies.
+        The CCID is a grouping of content as defined by <xref target="RFC8006"/>.
+        The "ccids" spec type is valid only for the content spec subject (see <xref target="trigger-subject"/>).
```

Comment:
The spec mentions that this is only valid for content spec subjects.  Should we also perhaps mention that CCIDs Spec, URI Pattern Match Spec, and URI Regex Match Spec are not valid or appropriate for preposition triggers, given the requirement to have well-defined URI's for prepositioning?

Reply:
added Spec constraints paragraph, also mentioned in my comments to Brett